lib/s3routes/routes/routePOST.ts

```diff
@@ -58,6 +58,15 @@ export default function routePOST(
                     corsHeaders));
     }
 
+    if (objectKey === undefined) {
+        if (Object.keys(query).length > 0) {
+            return routesUtils.responseNoBody(errors.InvalidArgument
+                .customizeDescription("Query String Parameters not allowed on POST requests."), null,
+                response, 400, log);
+        }
+        return api.callPostObject!('objectPost', request, response, log, (err, resHeaders) => routesUtils.responseNoBody(err, resHeaders, response, 204, log));
```

Comment:
non null not neded anymore
```suggestion
        return api.callPostObject('objectPost', request, response, log, (err, resHeaders) => routesUtils.responseNoBody(err, resHeaders, response, 204, log));
```